src/shared/Table/ResponsiveTable.tsx

```diff
@@ -60,12 +61,16 @@ export type ResponsiveTableProps<TRow, TCol> = {
   ) => (ResponsiveThProps["sort"] & { label: string }) | undefined;
   isRowDeleted?: (props: RowProps<TRow>) => boolean;
   isRowSelected?: (props: RowProps<TRow>) => boolean;
+  isRowChecked?: (rowIndex: number) => boolean;
   expectedLength?: number;
   onRowClick?: (props: RowProps<TRow>) => void;
   setActionCellOuiaId?: (props: RowProps<TRow>) => string;
   setRowOuiaId?: (props: RowProps<TRow>) => string;
   tableOuiaId?: string;
   variant?: TableVariant;
+  onCheck?: (isSelecting: boolean, rowIndex: number) => void;
```

Comment:
If you look at how the other callbacks are implemented, we pass the row object back to the callback, not an index. Eg. `isRowSelected?: (props: RowProps<TRow>) => boolean;`

This is because the index means nothing for who is using the table, as it depends on things like sorting and pagination that are too messy to deal with outside of the table itself.

Please correct it on this and the other new callbacks.